lib/request.js

```diff
@@ -41,8 +42,9 @@ function buildRequest (R, trustProxy) {
 }
 
 function buildRegularRequest (R) {
-  function _Request (id, params, req, query, log) {
+  function _Request (id, params, req, query, log, context) {
     this.id = id
+    this.context = context
```

Comment:
Good spot, sorry for the incovenience.

Reply:
No worries - my stuff here is all R&D still, so things break ALL THE TIME. :+1:

Reply:
same here. Please can we get the doc updated with this new prop in request?

Reply:
> same here. Please can we get the doc updated with this new prop in request?

PRs are welcome.

Reply:
> PRs are welcome.

@jsumners I took a brief look at the code and it seems that `Request#context` is an instance of `Context`.

https://github.com/fastify/fastify/blob/9d703b006bf797407d1ab50e6de6aa564fc2c0b4/lib/route.js#L233-L247

This object is not currently in the API docs. Do you suggest that the entire `Context` object be documented as part of the public API? Or instead, simply document portions of `request.context` that are relevant to a user. When I search the current docs for "context", I only find it mentioned in passing as a part of the `Reply` object.

https://github.com/fastify/fastify/blame/main/docs/Reply.md#L68-L76

It's odd, because the purpose of this PR seems to be solely about exposing `routerPath`, `routerMethod` and `is404` getter properties on `Request` by proxying these to the `context`. 

https://github.com/fastify/fastify/blob/9d703b006bf797407d1ab50e6de6aa564fc2c0b4/lib/route.js#L233-L247

It's just a side effect is that a `context` instance is now part of the public API, and I wonder if that's really what is intended since this PR did actually document what the expected API changes were. If it's intended, it's not clear what of the `Context` object should be publicly documented. If it's unintended, then it perhaps should not be documented? :shrug:

Reply:
I'm confused. This PR is merged and has doc changes that show the properties added in the request documentation. What is missing in the live docs?

Reply:
@jsumners that was my question! It's not clear... This change introduced a `context` property on the `Request` object. But that was not the intent of the change, and so was never documented. Back in September, when I upgraded to a version containing this change, my application broke because of this new, undocumented property. When I figured out what had happened, I came here to mention it. Three days ago, someone else came here to mention the same thing and asked for doc changes. Then you said PRs are welcome, so I figured I would do my part and look into it. But, as I've said, it's not clear what exactly should be done.

Reply:
It's _sort of_ documented under https://www.fastify.io/docs/latest/Routes/#config

@mcollina posted some reasoning in another PR discussion around why we don't fully document it. But I don't recall which.

Reply:
See https://github.com/fastify/fastify/pull/2989